src/components/ClusterNodes/NodeList.tsx

```diff
@@ -0,0 +1,526 @@
+        let _flattenNodeList = []
+        for (let index = 0; index < flattenNodeList.length; index++) {
+            const element = flattenNodeList[index]
+            if (selectedVersion.value !== defaultVersion.value && element['k8sVersion'] !== selectedVersion.value) {
+                continue
+            }
+            if (selectedSearchTextType === 'name' && element['name'].indexOf(searchText) === -1) {
+                continue
+            } else if (selectedSearchTextType === 'label') {
+                let matchedLabelCount = 0
+                for (let i = 0; i < element['labels']?.length; i++) {
+                    const currentLabel = element['labels'][i]
+                    const matchedLabel = searchedLabelMap.get(currentLabel.key)
+                    if (matchedLabel === undefined || (matchedLabel !== null && currentLabel.value !== matchedLabel)) {
+                        continue
+                    }
+                    matchedLabelCount++
+                }
+                if (searchedLabelMap.size !== matchedLabelCount) {
+                    continue
+                }
+            }
+            _flattenNodeList.push(element)
+        }
+        if (sortByColumn) {
+            const comparatorMethod =
+                sortByColumn.sortType === 'number' ? numericComparatorMethod : alphabeticalComparatorMethod
+            _flattenNodeList.sort(comparatorMethod)
+        }
+        setFilteredFlattenNodeList(_flattenNodeList)
+        setNoResults(_flattenNodeList.length === 0)
+    }
+
+    const numericComparatorMethod = (a, b) => {
+        let firstValue = a[sortByColumn.sortingFieldName] || 0
+        let secondValue = b[sortByColumn.sortingFieldName] || 0
+        if (typeof firstValue === 'string' && firstValue?.endsWith('%')) {
+            firstValue = firstValue.slice(0, -1)
+            secondValue = secondValue.slice(0, -1)
+        }
+        return sortOrder === OrderBy.ASC ? firstValue - secondValue : secondValue - firstValue
+    }
+
+    const alphabeticalComparatorMethod = (a, b) => {
+        return (sortOrder === OrderBy.ASC && sortByColumn.sortingFieldName !== 'createdAt') ||
+            (sortOrder === OrderBy.DESC && sortByColumn.sortingFieldName === 'createdAt')
+            ? a[sortByColumn.sortingFieldName].localeCompare(b[sortByColumn.sortingFieldName])
+            : b[sortByColumn.sortingFieldName].localeCompare(a[sortByColumn.sortingFieldName])
+    }
+
+    const clearFilter = (): void => {
+        setSearchText('')
+        setSelectedSearchTextType('')
+        setSearchedLabelMap(new Map())
+    }
+
+    useEffect(() => {
+        handleFilterChanges()
+    }, [searchedLabelMap, searchText, flattenNodeList, sortByColumn, sortOrder])
+
+    const onClusterChange = (selectedValue: OptionType): void => {
+        setSelectedCluster(selectedValue)
+        const currentUrl = match.url.replace(clusterId, selectedValue.value)
+        history.push(currentUrl)
```

Comment:
Directly pass `match.url.replace(clusterId, selectedValue.value)` to push. No need of creating a new var.

Reply:
Done